espnet2/asr/ctc.py

```diff
@@ -62,6 +68,59 @@ def loss_fn(self, th_pred, th_target, th_ilen, th_olen) -> torch.Tensor:
                 loss = loss / size
             return loss
 
+        elif self.ctc_type == "builtin2":
```

Comment:
can you add some clarification on builtin vs. builtin2?

Reply:
Builtin2 was the old way that espnet was handling nan in CTC. At some point, it was switched to using the zero_infinity flag in pytorch's CTC - I don't know the details of how they do that.

But I found that there are differences in the performance of resulting models. For instance, builtin2 was 0.3 bleu better on MuST-C en-de. This was the result we published in the EACL paper. I thought it might be better to keep builtin2 as an option.

I will add a description to the code.